apple/testing/default_runner/macos_test_runner.template.sh

```diff
@@ -137,8 +137,10 @@ fi
 # Run xcodebuild with the xctestrun file just created. If the test failed, this
 # command will return non-zero, which is enough to tell bazel that the test
 # failed.
+mkdir -p "${TEST_UNDECLARED_OUTPUTS_DIR}"
```

Comment:
```suggestion
rm -rf "$TEST_UNDECLARED_OUTPUTS_DIR/test.xcresult"
```

We shouldn't need to make this directory ahead of time, but there is an issue where it isn't cleaned if you have a flaky test that is rerun multiple times, and xcodebuild will fail if the path already exists